docs/pages/database-access/guides/postgres-cloudsql.mdx

```diff
@@ -0,0 +1,336 @@
+    # GCP specific configuration when connecting Cloud SQL instance.
+    gcp:
+      # GCP project ID.
+      project_id: "kubeadm-167321"
+      # Cloud SQL instance ID.
+      instance_id: "test"
+    # Labels to assign to the database, used in RBAC.
+    static_labels:
+      env: dev
+auth_service:
+  enabled: "no"
+ssh_service:
+  enabled: "no"
+proxy_service:
+  enabled: "no"
+```
+
+<Admonition
+  type="tip"
+  title="Tip"
+>
+  A single Teleport process can run multiple different services, for example
+  multiple database access proxies as well as running other services such an
+  SSH service or an application access proxy.
+</Admonition>
+
+Start the database service:
+
+```sh
+$ teleport start --config=/path/to/teleport-db.yaml --token=/tmp/token
+```
+
+### GCP Credentials
+
+Teleport Database Service must have credentials of `teleport-db-service` GCP
+service account we created [above](#create-service-account-for-teleport-database-service)
+in order to be able to generate IAM auth tokens.
+
+The easiest way to ensure that is to set `GOOGLE_APPLICATION_CREDENTIALS`
+environment variable to point to the JSON credentials file you downloaded
+earlier.
+
+See [Authenticating as a service account](https://cloud.google.com/docs/authentication/production)
+in Google Cloud documentation for more details.
+
+## Connect
+
+Once the database service has joined the cluster, login to see the available
+databases:
+
+```sh
+$ tsh login --proxy=teleport.example.com:3080 --user=testuser
+$ tsh db ls
+Name     Description              Labels
+-------- ------------------------ --------
+cloudsql GCP Cloud SQL PostgreSQL env=dev
+```
+
+Note that you will only be able to see databases your role has access to. See
+[RBAC](../rbac.mdx) section for more details.
+
+To connect to a particular database server, first retrieve credentials from
+Teleport using `tsh db login` command:
+
+```sh
+$ tsh db login cloudsql
+```
+
+<Admonition
+  type="tip"
+  title="Tip"
+>
+  You can be logged into multiple databases simultaneously.
+</Admonition>
+
+You can optionally specify the database name and the user to use by default
+when connecting to the database instance:
+
+```sh
+$ tsh db login --db-user=teleport@<project-id>.iam --db-name=postgres cloudsql
+```
+
+<Admonition
+  type="note"
+  title="What database user name to use?"
+>
+  When connecting to the database, use the name of the database service account
+  that you added as IAM database user [above](#create-service-account-for-database),
+  minus the `.gserviceaccount.com` suffix. The database user name is shown on
+  the Users page of your Cloud SQL instance.
+</Admonition>
+
+When logging into a PostgreSQL database, `tsh` automatically configures a section
+in the [connection service file](https://www.postgresql.org/docs/current/libpq-pgservice.html)
+with the name of `<cluster-name>-<database-service-name>`.
+
+Suppose the cluster name is "root", then you can connect to the database using
```

Comment:
When using `teleport configure` we now set cluster name as the FQDN of the cluster. To make the clsuter naming less confusing should we change this to Suppose the cluster name is "teleport.example.com" .... use `psql "service=teleport.example.com-cloudsql"` ?

Reply:
Yeah we can. Updated.